Resolve Errors - PR

soy-park opened the pull request:
All remaining errors are the same across files. These errors relate to extension and reference URLs, which I’ve attempted to resolve using alternative functional and relevant extensions and references. Detailed descriptions of these errors are provided below.

### Files with extension errors:
- [input/resources/library-FHIR-ModelInfo.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIR-ModelInfo.html)
- [input/resources/library-FHIRCommon.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRCommon.html)
- [input/resources/library-FHIRCommonTests.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRCommonTests.html)
- [input/resources/library-FHIRHelpers.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRHelpers.html) 
- [input/resources/library-FHIRHelpersTests.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRHelpersTests.html)
    - Error: The extension http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-softwaresystem could not be found so is not allowed here
    - Attempt: I tried to use the following functional extensions (below) with a similar valueReference to “Device/cqf-tooling”, but still getting an error
        - http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-softwaresystem
        - https://hl7.org/fhir/us/cqfmeasures/STU4/Device-software-system-example.html
        - https://hl7.org/fhir/us/cqfmeasures/STU4/StructureDefinition-cqfm-softwaresystem.htmlhttps://hl7.org/fhir/extensions/extensions-Device.html
        - http://terminology.hl7.org/ValueSet/software-system-type

### Files with reference errors: 
- [input/resources/library-FHIR-ModelInfo.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIR-ModelInfo.html)
- [input/resources/library-FHIRCommon.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRCommon.html) 
- [input/resources/library-FHIRCommonTests.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRCommonTests.html)
- [input/resources/library-FHIRHelpers.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRHelpers.html) 
- [input/resources/library-FHIRHelpersTests.json](https://github.com/cqframework/cqf/compare/master...soy-park:cqf:fix/Library-FHIRHelpersTests.html)
    - Error: Unable to resolve resource with reference 'http://hl7.org/fhir/us/cqfmeasures/STU3/Device-software-system-example.html'
    - I have already tried the following as valueReference, but still getting an error: 
        - Device/cqf-tooling
        - https://build.fhir.org/ig/HL7/fhir-mmm-ig/Device-cqf-tooling.html
        - http://hl7.org/fhir/us/cqfmeasures/STU3/Device-software-system-example.html

soy-park committed:
Add large files tracked with LFS

soy-park committed:
Removed input-cache files

soy-park committed:
Removed input-cache files

soy-park committed:
Merge branch 'master' into fix/adress-errors

soy-park committed:
Merge branch 'fix/adress-errors' of https://github.com/soy-park/cqf into fix/adress-errors

soy-park committed:
Merge remote-tracking branch 'upstream/master' into fix/adress-errors

brynrhodes commented on .gitattributes:
```diff
@@ -0,0 +1 @@
+*.jar filter=lfs diff=lfs merge=lfs -text
```
Why these lfs attributes? What problem is being addressed by using this attribute?

soy-park replied:
This file was automatically created when I installed Large File Storage. The command terminal suggested I install LFS because the repository was too large and slow.

soy-park replied:
I went ahead and removed (or at least, commented out for now) .gitattributes file to keep the repository more consistent.

brynrhodes commented on _gencontinuous.sh:
Why is this empty? This should just be the scripts as they are updated by the _updatePublisher script?

soy-park replied:
If you click on "View file," I see the following script on my end: 
```
#!/bin/bash
./_genonce.sh -watch
```
It may display as an "Empty file" because of whitespace changes or there's no difference from the master branch? Let me know if it's still not visible.

brynrhodes commented on _genonce.sh:
Also this one?

soy-park replied:
In "View file", it shows the following content: 
```
#!/bin/bash
publisher_jar=publisher.jar
input_cache_path=./input-cache/
echo Checking internet connection...
curl -sSf tx.fhir.org > /dev/null

if [ $? -eq 0 ]; then
	echo "Online"
	txoption=""
else
	echo "Offline"
	txoption="-tx n/a"
fi

echo "$txoption"

export JAVA_TOOL_OPTIONS="$JAVA_TOOL_OPTIONS -Dfile.encoding=UTF-8"

publisher=$input_cache_path/$publisher_jar
if test -f "$publisher"; then
	java -jar $publisher -ig . $txoption $*

else
	publisher=../$publisher_jar
	if test -f "$publisher"; then
		java -jar $publisher -ig . $txoption $*
	else
		echo IG Publisher NOT FOUND in input-cache or parent folder.  Please run _updatePublisher.  Aborting...
	fi
fi
```

brynrhodes commented on _updatePublisher.sh:
And this one?

soy-park replied:
This one appears to have content as well: 
```
#!/bin/bash
pubsource=https://github.com/HL7/fhir-ig-publisher/releases/latest/download/
publisher_jar=publisher.jar
dlurl=$pubsource$publisher_jar

input_cache_path=$PWD/input-cache/

scriptdlroot=https://raw.githubusercontent.com/HL7/ig-publisher-scripts/main
update_bat_url=$scriptdlroot/_updatePublisher.bat
gen_bat_url=$scriptdlroot/_genonce.bat
gencont_bat_url=$scriptdlroot/_gencontinuous.bat
gencont_sh_url=$scriptdlroot/_gencontinuous.sh
gen_sh_url=$scriptdlroot/_genonce.sh
update_sh_url=$scriptdlroot/_updatePublisher.sh

skipPrompts=false
FORCE=false

if ! type "curl" > /dev/null; then
	echo "ERROR: Script needs curl to download latest IG Publisher. Please install curl."
	exit 1
fi

while [ "$#" -gt 0 ]; do
    case $1 in
    -f|--force)  FORCE=true ;;
    -y|--yes)  skipPrompts=true ; FORCE=true ;;
    *)  echo "Unknown parameter passed: $1.  Exiting"; exit 1 ;;
    esac
    shift
done

echo "Checking internet connection"
curl -sSf tx.fhir.org > /dev/null

if [ $? -ne 0 ] ; then
  echo "Offline (or the terminology server is down), unable to update.  Exiting"
  exit 1
fi

if [ ! -d "$input_cache_path" ] ; then
  if [ $FORCE != true ]; then
    echo "$input_cache_path does not exist"
    message="create it?"
    read -r -p "$message" response
    else
    response=y
  fi
fi

if [[ $response =~ ^[yY].*$ ]] ; then
  mkdir ./input-cache
fi

publisher="$input_cache_path$publisher_jar"

if test -f "$publisher" ; then
	echo "IG Publisher FOUND in input-cache"
	jarlocation="$publisher"
	jarlocationname="Input Cache"
	upgrade=true
else
	publisher="../$publisher_jar"
	upgrade=true
	if test -f "$publisher"; then
		echo "IG Publisher FOUND in parent folder"
		jarlocation="$publisher"
		jarlocationname="Parent Folder"
		upgrade=true
	else
		echo "IG Publisher NOT FOUND in input-cache or parent folder"
		jarlocation=$input_cache_path$publisher_jar
		jarlocationname="Input Cache"
		upgrade=false
	fi
fi

if [[ $skipPrompts == false ]]; then

  if [[ $upgrade == true ]]; then
    message="Overwrite $jarlocation? (Y/N) "
  else
    echo Will place publisher jar here: "$jarlocation"
    message="Ok (enter 'y' or 'Y' to continue, any other key to cancel)?"
  fi
  read -r -p "$message" response
else
  response=y
fi
if [[ $skipPrompts == true ]] || [[ $response =~ ^[yY].*$ ]]; then

	echo "Downloading most recent publisher to $jarlocationname - it's ~100 MB, so this may take a bit"
	curl -L $dlurl -o "$jarlocation" --create-dirs
else
	echo cancelled publisher update
fi

if [[ $skipPrompts != true ]]; then
    message="Update scripts? (enter 'y' or 'Y' to continue, any other key to cancel)?"
    read -r -p "$message" response
  fi

if [[ $skipPrompts == true ]] || [[ $response =~ ^[yY].*$ ]]; then
  echo "Downloading most recent scripts "

  curl -L $update_bat_url -o /tmp/_updatePublisher.new
  cp /tmp/_updatePublisher.new _updatePublisher.bat
  rm /tmp/_updatePublisher.new

  curl -L $gen_bat_url -o /tmp/_genonce.new
  cp /tmp/_genonce.new _genonce.bat
  rm /tmp/_genonce.new

  curl -L $gencont_bat_url -o /tmp/_gencontinuous.new
  cp /tmp/_gencontinuous.new _gencontinuous.bat
  rm /tmp/_gencontinuous.new

  curl -L $gencont_sh_url -o /tmp/_gencontinuous.new
  cp /tmp/_gencontinuous.new _gencontinuous.sh
  chmod +x _gencontinuous.sh
  rm /tmp/_gencontinuous.new

  curl -L $gen_sh_url -o /tmp/_genonce.new
  cp /tmp/_genonce.new _genonce.sh
  chmod +x _genonce.sh
  rm  /tmp/_genonce.new

  curl -L $update_sh_url -o /tmp/_updatePublisher.new
  cp /tmp/_updatePublisher.new _updatePublisher.sh
  chmod +x _updatePublisher.sh
  rm /tmp/_updatePublisher.new
fi
```

brynrhodes commented on input/resources/Library-FHIR-ModelInfo.json:
```diff
@@ -13,6 +13,11 @@
+    "valueReference": {
+      "reference": "http://hl7.org/fhir/us/cqfmeasures/STU3/Device-software-system-example.html"
+    }
   } ],
```
For the CQF, I just removed all the cqf-tooling stuff, so we shouldn't add this back in

brynrhodes commented on input/resources/device/cqf-tooling.json:
This file shouldn't be added, I'm not using the software system device stuff in this IG

soy-park committed:
feat: Update warnings to ignore in ignoreWarnings.txt

soy-park committed:
feat: Update xml files in pagecontent directory

soy-park committed:
feat: Remove url and value for cqf-tooling

soy-park committed:
feat: Remove 'input/resources/device/cqf-tooling.json' file

soy-park committed:
Remove input-cache files and cqf tooling/software system url

soy-park committed:
Remove LFS tracking

soy-park commented:
Requested changes should now be resolved. 
- I removed (or at least, commented out for now) .gitattributes file
- For "empty" files, there appears to be content if you click on "View file." I also made sure the appropriate scripts were in each .sh file
- I also removed software system device content and related cqf-tooling
